# PN10B event-centred geometry diagnostic

## tl;dr

The frozen PN10B predictive verdict remains **NULL**, but that verdict does not mean the geometry was empty. At a prime node the parent factor-survival coordinate reaches an exact **1.0 ridge**, with immediate odd offsets dropping to the even-number trough near **0.062701**. Inside the node, the nine paid-gate children remain strongly asymmetric: their individual A readings span almost the full 0–2 line, and the first prime's vector runs from **0.071972** to **1.435762**. Across 45,166 primes those child readings aggregate to **0.999861**, but survivor composites aggregate to **0.998614**, so the child geometry is real while its prime-specific separation is negligible in this representation.

This notebook is a **post-hoc descriptive disclosure**, not a fresh prediction test.

## Context & Methods

The fresh interval is `[4,000,000,000, 4,001,000,000)`. PN10B had already defined nine paid-gate children using the largest nine prime gates `q <= n^0.45`:

- `A_j(n) = 2 (n mod q_j) / q_j`
- `B_j(n) = 2 - A_j(n)`
- child centroid: mean of the nine A readings
- child dispersion: mean absolute distance from the 1.0 ridge
- adjacent coupling: mean `(A_j-1)(A_{j+1}-1)`
- parent factor progress: `1` for a prime, otherwise `2 log(LPF(n)) / log(n)`

### Key Assumptions

The parent coordinate is a factor-survival description, not an advance forecast: a prime receives 1.0 only after surviving every required divisor test through its square root. The child coordinate is the paid-gate proxy tested in PN10B; it may not be the final internal child definition intended by the full ARA ontology.


In [1]:
from pathlib import Path
import csv, json

ROOT = Path(r'F:\SystemFormulaFolder\GIT\ARA-GIT\analysis\primes')
results = json.loads((ROOT / 'PN10B_EVENT_GEOMETRY_RESULTS.json').read_text(encoding='utf-8'))
validation = json.loads((ROOT / 'PN10B_EVENT_GEOMETRY_VALIDATION.json').read_text(encoding='utf-8'))
with (ROOT / 'PN10B_EVENT_CENTERED_TRACES.csv').open(newline='', encoding='utf-8') as handle:
    traces = list(csv.DictReader(handle))
with (ROOT / 'PN10B_CHILD_LANDMARK_COUNTS.csv').open(newline='', encoding='utf-8') as handle:
    landmarks = list(csv.DictReader(handle))
with (ROOT / 'PN10B_PRIME_CHILD_EXAMPLES.csv').open(newline='', encoding='utf-8') as handle:
    examples = list(csv.DictReader(handle))
print('## Data')
print(json.dumps(results['scope'], indent=2))
print('Independent validation:', validation['checks_passed'], '/', validation['checks_total'], 'passed=', validation['passed'])


## Data
{
  "low_inclusive": 4000000000,
  "high_exclusive": 4001000000,
  "raw_integer_count": 1000000,
  "prime_count": 45166,
  "c090_survivor_count": 54275,
  "c090_survivor_composite_count": 9109,
  "child_gate_count": 9,
  "event_window_each_side": 32
}
Independent validation: 12 / 12 passed= True


In [2]:
print('## Results — population geometry')
for population in ('prime', 'survivor_composite'):
    node = results['node_distributions'][population]
    print(population)
    for metric in ('child_centroid', 'child_dispersion', 'child_coupling', 'child_flip_count', 'pooled_child_phase_a'):
        row = node[metric]
        print(f"  {metric:<22} mean={row['mean']:.9f} sd={row['sd']:.9f} median={row['median']:.9f} range=[{row['min']:.9f},{row['max']:.9f}]")
print('standardized prime-minus-composite differences')
for metric, row in results['population_contrasts'].items():
    print(f"  {metric:<22} raw={row['prime_minus_survivor_composite_mean']:+.9f} standardized={row['standardized_mean_difference']:+.6f}")


## Results — population geometry
prime
  child_centroid         mean=0.999860546 sd=0.200419075 median=1.032589083 range=[0.499788889,1.426638535]
  child_dispersion       mean=0.499516557 sd=0.093933345 median=0.499199523 range=[0.334814336,0.665169823]
  child_coupling         mean=-0.014075849 sd=0.118656249 median=0.011770684 range=[-0.193552588,0.258944518]
  child_flip_count       mean=4.128636585 sd=1.504680011 median=4.000000000 range=[1.000000000,6.000000000]
  pooled_child_phase_a   mean=0.999860546 sd=0.576870871 median=0.999856370 range=[0.000095516,1.999904439]
survivor_composite
  child_centroid         mean=0.998614374 sd=0.200236629 median=1.034945972 range=[0.503233374,1.422809661]
  child_dispersion       mean=0.499724107 sd=0.093512643 median=0.499382752 range=[0.334857071,0.665137658]
  child_coupling         mean=-0.015149291 sd=0.118753726 median=0.009703038 range=[-0.193528941,0.255498254]
  child_flip_count       mean=4.150290921 sd=1.490086969 median=4.00000000

In [3]:
print('## Results — event-centred lead / at / lag')
selected = {-10,-9,-8,-7,-6,-5,-4,-3,-2,-1,0,1,2,3,4,5,6,7,8,9,10}
print('offset | prime rate | parent progress | child centroid | dispersion | coupling | flips')
for row in traces:
    if row['event'] == 'prime_center' and int(row['offset']) in selected:
        print(f"{int(row['offset']):>+3} {float(row['prime_rate']):11.6f} {float(row['parent_progress_mean']):16.9f} "
              f"{float(row['child_centroid_mean']):16.9f} {float(row['child_dispersion_mean']):11.9f} "
              f"{float(row['child_coupling_mean']):+11.9f} {float(row['child_flip_count_mean']):7.4f}")


## Results — event-centred lead / at / lag
offset | prime rate | parent progress | child centroid | dispersion | coupling | flips
-10    0.077632      0.286799820      0.999813003 0.499508349 -0.014115130  4.1290
 -9    0.000000      0.062700773      0.999790614 0.499508327 -0.014100459  4.1289
 -8    0.059497      0.246121268      0.999773146 0.499508306 -0.014098013  4.1289
 -7    0.000000      0.062700773      0.999755678 0.499508282 -0.014078722  4.1289
 -6    0.121075      0.393390852      0.999757893 0.499508256 -0.014067189  4.1286
 -5    0.000000      0.062700773      0.999740425 0.499508231 -0.014065481  4.1286
 -4    0.059519      0.247668081      0.999747560 0.499508205 -0.014086528  4.1288
 -3    0.000000      0.062700773      0.999774377 0.499508181 -0.014078300  4.1287
 -2    0.059475      0.246651626      0.999796273 0.499508158 -0.014086227  4.1286
 -1    0.000000      0.062700773      0.999788646 0.499508139 -0.014073417  4.1286
 +0    1.000000      1.000000000      0.

In [4]:
print('## Results — exact first-prime child vector')
print('rank q remainder A B signed_A_minus_1 coupling_to_next')
for row in examples:
    if row['example'] == 'first_prime':
        print(row['gate_rank'], row['gate_q'], row['remainder'], row['phase_a'], row['phase_b'],
              row['signed_orientation'], row['coupling_to_next_rank'])


## Results — exact first-prime child vector
rank q remainder A B signed_A_minus_1 coupling_to_next
1 20929 7669 0.7328587127908643 1.2671412872091357 -0.26714128720913566 0.026776697064077128
2 20921 9412 0.8997657855743033 1.1002342144256967 -0.10023421442569669 0.08175348618589212
3 20903 1927 0.18437544850021528 1.8156245514997846 -0.8156245514997847 -0.3554185745972793
4 20899 15003 1.435762476673525 0.5642375233264749 0.4357624766735251 -0.4043997564305724
5 20897 752 0.07197205340479494 1.928027946595205 -0.928027946595205 -0.33247633094134715
6 20887 14185 1.3582611193565375 0.6417388806434625 0.3582611193565375 -0.3175258400159359
7 20879 1187 0.11370276354231525 1.8862972364576847 -0.8862972364576848 0.6610818605092557
8 20873 2652 0.25410817802903274 1.7458918219709672 -0.7458918219709673 0.5432633009330663
9 20857 2833 0.2716593949273625 1.7283406050726375 -0.7283406050726375 


In [5]:
print('## Results — landmark occupancy shares')
for row in landmarks:
    print(f"{row['population']:<20} {row['landmark_region']:<27} {float(row['share']):.9f}")
print('Static figure:', ROOT / 'PN10B_EVENT_GEOMETRY_FIGURE.png')


## Results — landmark occupancy shares
prime                left_singularity_well       0.124651286
prime                left_inner_to_phi           0.066328162
prime                left_phi_to_ridge           0.309106653
prime                ridge_to_right_phi          0.309118954
prime                right_phi_to_inner          0.066618449
prime                right_singularity_well      0.124176495
survivor_composite   left_singularity_well       0.124358083
survivor_composite   left_inner_to_phi           0.066271453
survivor_composite   left_phi_to_ridge           0.309376563
survivor_composite   ridge_to_right_phi          0.310279211
survivor_composite   right_phi_to_inner          0.066576402
survivor_composite   right_singularity_well      0.123138288
Static figure: F:\SystemFormulaFolder\GIT\ARA-GIT\analysis\primes\PN10B_EVENT_GEOMETRY_FIGURE.png


## Takeaways

1. **The prime event is a sharp parent ridge.** The parent factor-progress coordinate is 1.0 at the prime and falls into the parity-driven trough on every odd offset around an odd prime.
2. **The internal paid-gate children are not quiet.** A single prime can contain large A/B asymmetries and several sign flips even when population aggregation sits near 1.0.
3. **The population ridge is cancellation across nodes.** Pooled prime child A is 0.999861, yet individual prime centroids range from 0.499789 to 1.426639.
4. **That child pattern is not prime-specific here.** Survivor composites have almost the same centroid, dispersion, coupling, flip-count and landmark distributions. All standardized mean differences are below 0.015.
5. **Therefore both statements must be reported.** PN10B is NULL as a prime-ranking test, while its post-hoc geometry shows an exact parent event ridge and rich but non-discriminating child waves.

The next confirmatory step, if pursued, must freeze what counts as the prime node's *internal* Phase A and Phase B before opening a new interval.
